test/vfd_swmr_dsetchks_writer.c

```diff
@@ -0,0 +1,2269 @@
+        if (!s->implicit_index && s->ydecrs) {
+            for (step = 0; step < s->ydecrs; step++) {
+                dbgf(2, "Verify decrease to dimension sizes by %u for datasets with fa/ea/bt2 index\n",
+                     step + 1);
+
+                if (s->use_np && !np_confirm_verify_notify(np->fd_writer_to_reader, step, s, np)) {
+                    printf("np_confirm_verify_notify() verify/notify not in sync failed\n");
+                    TEST_ERROR;
+                }
+                /* Wait for a few ticks for the update to happen */
+                decisleep(config->tick_len * s->update_interval);
+
+                result = verify_dsets_extent(DECR_EXT, s, ds, step + 1);
+
+                if (s->use_np && !np_reader(result, step, s, np)) {
+                    printf("np_reader() for verification failed\n");
+                    TEST_ERROR;
+                }
+            }
+        }
+    }
+
+    return true;
+
+error:
+
+    return false;
+
+} /* verify_dsets_operations() */
+
+/*
+ * Verify the data read from each of the specified datasets:
+ *      GWRITES: verify `which` write that covers a single chunk
+ *      PWRITES: verify `which` write that covers a partial chunk
+ *      TWRITES: verify `which` write that covers multiple chunks
+ *      LWRITEs: verify `which` write that covers multiple partial chunks
+ */
+static bool
+verify_dsets_chunks(unsigned action, const state_t *s, const dsets_state_t *ds, unsigned which)
+{
+    hsize_t       start[2]  = {0, 0};
+    hsize_t       stride[2] = {0, 0};
+    hsize_t       count[2]  = {0, 0};
+    hsize_t       block[2]  = {0, 0};
+    unsigned int *vbuf      = NULL;
+
+    HDassert(s->implicit_index || s->fa_index || s->ea_index || s->bt2_index);
+
+    /* Set up selection according to the specified action */
+    setup_selection(action, which, s, ds, start, stride, count, block);
+
+    if (s->implicit_index) {
+        if (!verify_read_dset(action, ds->implicit_did, s->filetype, start, stride, count, block)) {
+            printf("verify_read_dset() to dataset with implicit index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (s->fa_index) {
+        if (!verify_read_dset(action, ds->fa_did, s->filetype, start, stride, count, block)) {
+            printf("verify_read_dset() to dataset with fixed array index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (s->ea_index) {
+        if (!verify_read_dset(action, ds->ea_did, s->filetype, start, stride, count, block)) {
+            printf("verify_read_dset() to dataset with extensible array index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (s->bt2_index) {
+        if (!verify_read_dset(action, ds->bt2_did, s->filetype, start, stride, count, block)) {
+            printf("verify_read_dset() to dataset with bt2 index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (vbuf)
```

Comment:
vbuf is freed here, but it is never used.

Reply:
corrected